packages/esm-patient-forms-app/src/forms/forms-see-all.component.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import first from 'lodash-es/first';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { fhir } from '@openmrs/esm-framework';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+interface FormsProps {
+  patientUuid: string;
+  patient: fhir.Patient;
+}
+
+const Forms: React.FC<FormsProps> = ({ patientUuid, patient }) => {
+  async function setPatient(uUid) {
+    patient = (await fhir.read<fhir.Patient>({ type: 'Patient', patient: uUid })).data;
```

Comment:
This is fishy on multiple levels.

Please rewrite using a side effect and changing a proper state.

Reply:
Sure. I had adopted this from a different component but there is a need to optimise it further. I will make the necessary corrections.

Reply:
I have now removed the earlier function and replaced it with a 'State Hook'

Reply:
Optimised this further with Effect Hook